TheMET/TheMET/Network/MetAPI.swift

```diff
@@ -106,42 +108,43 @@ class MetAPI{
             parameters: [ : ]) { result in
                 switch result {
                 case .failure:
-                    completion(nil)
+                    completion(.failure(.invalidUrlComponents))
                 case .success(let data):
                     let jsonDecoder = JSONDecoder()
                     jsonDecoder.keyDecodingStrategy = .useDefaultKeys
                     do {
                         let result = try jsonDecoder.decode(ObjectResponse.self, from: data)
-                        completion(result)
+                        completion(.success(result))
                     } catch {
-                        completion(nil)
+                        completion(.failure(.noDataInResponse))
```

Comment:
This is not a `noDataInResponse` error. This error happens because the object-response can't be decoded from data. I suggest `nonDecodableData` name for this error